### DISCUSSION TITLE
com.arcadedb.console.ConsoleException: Database appears locked by server

### DISCUSSION THREAD

**Discussion by kalisitonga** [ArcadeData/arcadedb#1227] [Q&A] [answered]
Hi, I'm new to using arcadedb. I started up a docker container and created a database. I connected to it using the browser console and then using a terminal. In the terminal, after running "connect database test", I get this error:

ERROR:
com.arcadedb.console.ConsoleException: Database appears locked by server
        at com.arcadedb.console.Console.checkDatabaseIsLocked(Console.java:896)
        at com.arcadedb.console.Console.executeConnect(Console.java:410)
        at com.arcadedb.console.Console.execute(Console.java:249)
        at com.arcadedb.console.Console.parse(Console.java:731)
        at com.arcadedb.console.Console.interactiveMode(Console.java:129)
        at com.arcadedb.console.Console.execute(Console.java:179)
        at com.arcadedb.console.Console.main(Console.java:142)
        
I created a new database in the web console and tried switching to that in case that was locking it, but I still get the error in the terminal. In case it helps, I ran "lslocks" and get this:

java      1 POSIX      WRITE 0     0   0 /home/arcadedb/databases/test/database.lck

Also in case it helps, I get these warnings when opening the console in a terminal:
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.jline.terminal.impl.exec.ExecTerminalProvider (file:/home/arcadedb/lib/jline-terminal-3.23.0.jar) to constructor java.lang.ProcessBuilder$RedirectPipeImpl()
WARNING: Please consider reporting this to the maintainers of org.jline.terminal.impl.exec.ExecTerminalProvider
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release

I couldn't find anything in the documentation or in this discussion group. Is there something I should be doing to be able to connect to it in the command line console?

**Comment by gramian** [comment, marked as answer]
Indeed, this behavior is not yet documented. I added an issue to the docs repo: https://github.com/ArcadeData/arcadedb-docs/issues/190 , I will add a note about this to the docs next week.

AFAIK, when the server is running and opened a database (by default it opens all), the console is locked out.

About the second issue: when I ran into this once, @lvca suggested to append:
```
 --add-opens java.base/java.lang=ALL-UNNAMED
```
which help for me.

**Reply by kalisitonga** [reply]
Thank you for the quick help! Just found a workaround. If I go into the database folder under /databases/test and delete the database.lck file, it lets me connect to it using the console. I don't know it that's ok to do but it seems to be working for me.

Also, your suggestion on the 2nd issue worked for me. In case it helps anyone else:
docker run -d -p 2480:2480 -p 2424:2424 --env JAVA_OPTS="-Darcadedb.server.rootPassword=playwithdata **--add-opens java.base/java.lang=ALL-UNNAMED"** arcadedata/arcadedb:latest

**Reply by gramian** [reply]
I cannot generally recommend your workaround for the first issue. Deleting the lock file to obtain access via the console probably means write guarantees are void, but I assume you are aware of these potential consequences.

**Reply by lvca** [reply]
That's correct: the lock file is to protect the files from being overwritten by another ArcadeDB server instance or embedded. You could either:
- start one server and connect via remote (HTTP) protocol or
- create an app that uses the database in embedded mode and then it starts a server in the same JVM, so concurrent remote accesses are still possible

**Reply by kalisitonga** [reply]
Got it, I won't do that then :) Thanks for the help!
